oteljava/context-storage/src/main/scala/org/typelevel/otel4s/oteljava/context/IOLocalContextStorage.scala

```diff
@@ -75,34 +75,34 @@ object IOLocalContextStorage {
     * }
     *   }}}
     */
-  def localProvider[F[_]: LiftIO](implicit F: MonadCancelThrow[F]): LocalProvider[F, Context] =
+  def localProvider[F[_]: Console: LiftIO](implicit F: MonadCancelThrow[F]): LocalProvider[F, Context] =
     new LocalProvider[F, Context] {
       def local: F[Local[F, Context]] =
         ContextStorage.get() match {
           case storage: IOLocalContextStorage =>
             whenPropagationEnabled(storage.local)
 
-          case other if other.getClass.getName == AgentContextStorageClass =>
-            whenPropagationEnabled(agentLocal)
-
           case other =>
-            F.raiseError(
-              new IllegalStateException(
-                "IOLocalContextStorage is not configured for use as the ContextStorageProvider. " +
-                  s"The current storage: ${other.getClass.getName}."
-              )
-            )
+            getAgentLocalContext() match {
+              case Some(ioLocalJContext) =>
+                val ioLocal = ioLocalJContext.lens(ctx => Context.wrap(ctx))(_ => c => c.underlying)
+                val local = LocalProvider.localForIOLocal(ioLocal)
+
+                for {
+                  _ <- Console[F].println("IOLocalContextStorage: agent-provided IOLocal is detected")
+                  r <- whenPropagationEnabled(local)
+                } yield r
+
+              case None =>
+                F.raiseError(
+                  new IllegalStateException(
+                    "IOLocalContextStorage is not configured for use as the ContextStorageProvider. " +
+                      s"The current storage: ${other.getClass.getName}."
```

Comment:
this is a bit of a wording nitpick/bikeshed, but I think this should be one of the following:

```suggestion
                      s"The current ContextStorage is: ${other.getClass.getName}"
```

```suggestion
                      s"Current ContextStorage: ${other.getClass.getName}"
```